### ISSUE TITLE
Color shift in graphics converting LaTeX-document 

### ISSUE THREAD

**Issue by ka-bo** [matteosecli/pdf2archive#13]
Hello. 

I tried to convert a LaTeX generated pdf to pdfa using this tool. Unfortunately, the pdf-graphics I used in the document shifts in color a tiny bit and, e.g., gets a green hue instead of the original white background. I use gs version 9.52. Has this been experienced before?

Thanks

**Comment by matteosecli** [comment]
Hi @ka-bo, it's probably some issue with transparency (it has to be removed, because PDF/A doesn't support it, so colors *do* change in some cases) or with color conversion. Is your pdf graphics in RGB or in CMYK? If it's in CMYK, can you please try exporting in in RGB first, include it in your LaTeX and then run the conversion again?

The other thing is that I should really update this tool to reflect new changes in Ghostscript, but I've no time to do it. 🙂

If the above doesn't work, can you please try to replace these lines here:

https://github.com/matteosecli/pdf2archive/blob/076dae2b6b048b9b97354072bdb31fc66c50d9c1/pdf2archive#L349-L356

with these lines below?

```bash
run gs $MSGOPTS \
    -dPDFA=1 -dBATCH -dNOPAUSE \
    $QUALITYOPTS \
    -dPDFACompatibilityPolicy=1 \
    -dConvertCMYKImagesToRGB=true \
    -sProcessColorModel=DeviceRGB -sColorConversionStrategy=UseDeviceIndependentColor \
    -sOutputICCProfile=$ICCTMPFILE \
    -sDEVICE=pdfwrite \
    -sOutputFile=$OUTPUT $TMPFILE $PSTMPFILE
```

If that still doesn't work, can you please try by changing `UseDeviceIndependentColor` to `sRGB` or `RGB`, i.e. with these lines?

```bash
run gs $MSGOPTS \
    -dPDFA=1 -dBATCH -dNOPAUSE \
    $QUALITYOPTS \
    -dPDFACompatibilityPolicy=1 \
    -dConvertCMYKImagesToRGB=true \
    -sProcessColorModel=DeviceRGB -sColorConversionStrategy=sRGB \
    -sOutputICCProfile=$ICCTMPFILE \
    -sDEVICE=pdfwrite \
    -sOutputFile=$OUTPUT $TMPFILE $PSTMPFILE
```

-----

It would also help to have a copy of the PDF you're trying to convert, maybe just the page with the graphics or just the graphics itself (but the whole page would be better, as the conversion is done per-page and there might be something else messing up with the conversion of that specific page, resulting in color artifacts). If you have sensitive info in the file and you cannot share it here, feel free to send me a PM [here](mailto:secli.matteo@gmail.com).

**Comment by ka-bo** [comment]
Thanks for the quick and extensive reply.

I converted all the PDF images with your tool before building the LaTeX pdf, so they should be in RGB, right? Unfortunately, both changes you suggested also did not work for me. Probably you are right and it has something to do with transparency in the figures.

Meanwhile I converted the document using another (commercial) tool, resulting in a PDF-A/2b, which is sufficient for me.
